src/brogue/PowerTables.c

```diff
@@ -197,7 +197,8 @@ fixpt defenseFraction(fixpt netDefense) {
         469, 453, 439, 425, 411, 398, 385, 373, 361, 349, 338, 327, 316, 306, 296, 287, 277, 268, 260, 251, 243, 235, 228, 221, 213, 207,
         200, 193, 187, 181, 175, 170, 164, 159, 154, 149, 144, 139, 135, 130, 126, 122, 118, 114, 111, 107, 104, 100, 97, 94};
 
-    short idx = clamp(netDefense * 4 / 10 / FP_FACTOR + 80, 0, LAST_INDEX(POW_DEFENSE_FRACTION));
+    // Adding "FP_FACTOR * 5 / 4" here is necessary to ensure that an armor rating of 3.75 is different from 3.5. It doesn't change the result for armor ratings that are multiples of 0.5, but increases the index by 1 for armor ratings that end in 0.25 or 0.75.
+    short idx = clamp((netDefense + FP_FACTOR * 5 / 4) * 4 / 10 / FP_FACTOR + 80, 0, LAST_INDEX(POW_DEFENSE_FRACTION));
```

Comment:
This is giving me slightly bad vibes. Gone from adding 0 to 5/4. Busy now but I need to have another look later.

(If in debugging this issue you wished you could redo the entire armor system, I support you in this, seems some strange code)

Reply:
I added a long comment explaining what this code does (or trying to, anyway). I also modified this line to replace the ugly 5/4 constant with 0.5, while keeping the same behavior. It's still not perfect, but hopefully it's understandable now.

Reply:
Is there a gameplay issue here? If so, how would one duplicate it in wizard mode? I'm not keen on the UI change.

Reply:
> Is there a gameplay issue here? If so, how would one duplicate it in wizard mode? I'm not keen on the UI change.

The issue is that armor increments of 0.25 don't affect enemy hit chance (but increments of 0.5 do). To repro, create an enemy goblin (or similar) and several potions of strength, and check the monster's hit chance in between drinking strength potions. Only every other strength potion has an effect on its hit chance. This is inconsistent with your own hit chance, which increases slightly with each strength potion.

What don't you like about the UI change?

Reply:
Ah, ok that makes sense. On the UI changes, it's the change to the armor value in the info panel (top left) that I'm resistant to. There's a satisfying simplicity to the integer value.

I'm not opposed to changing the item details screen though, both for the equipped armor and when comparing an armor in inventory to the equipped armor.

Reply:
I'm not sure I like the idea of displaying the armor value with different precision in different places. I see your point about integers being simpler, but I think we should look for ways to keep the UI simple without putting the non-rounded armor value on a separate screen.

What about multiplying all displayed armor values by 4? The starting armor value would be 3.5*4 = 14, plate armor at 19 strength would have an armor value of 44, etc. Players would need to re-learn what armor values mean, though; I don't know whether it's worth it.